factory-containers/assemble-system-image

```diff
@@ -201,32 +224,32 @@ def copy_container_images_to_wic(target: dict, app_image_dir: str, wic_image: st
     else:
         app_image_tar_src = os.path.join(app_image_dir, containers_sha,
                                          '{}-{}.tar'.format(containers_sha, containers_arch))
+
+    # in kilobytes
+    image_data_size = None
     if not os.path.exists(app_image_tar_src):
         logger.info('Container images have not been found, trying to obtain them...')
         os.makedirs(os.path.dirname(app_image_tar_src), exist_ok=True)
-        dump_apps_container_images(target, containers_arch, app_image_tar_src, token)
-
-    # There is no guarantee that in the case of a `dump_apps_container_images` failure
-    # an exception or exit() will be called (some rough edge cases).
-    # Moreover, the following call `subprocess.check_call([wic_tool, 'cp', app_image_tar_src, app_image_tar_dst])`
-    # does not raise any exception if `app_image_tar_src` doesn't exist
-    # Therefore, in order to detect such subtle/silent error the following check is enabled,
-    # it just throws an exception if the tar file does not exist prior to injecting it to the target wic file.
-
-    try:
-        st = os.stat(app_image_tar_src)
-    except FileNotFoundError:
-        raise Exception('Container images archive {} has not been obtained'.format(app_image_tar_src))
-
-    # find size of the tar and increase the wic image
-    logger.info('Adding %d bytes to wic size', st.st_size)
-    resize_wic(wic_image, st.st_size)
-
-    app_image_tar_dst = '{}:{}{}'.format(wic_image, partition, directory)
-
-    logger.info('Copying container images of Target apps to its WIC image: {} --> {}'
-                .format(app_image_tar_src, app_image_tar_dst))
-    subprocess.check_call([wic_tool, 'cp', app_image_tar_src, app_image_tar_dst])
+        image_data_size = dump_apps_container_images(target, containers_arch, app_image_tar_src, token)
+    else:
+        with tempfile.TemporaryDirectory() as tmp_image_data:
+            # container images data have been already dumped and stored on NFS
+            # TODO: store the image data size on NFS along with the tarball
+            #  to avoid calling tar -xf <> to determine uncompressed image data size
+            str(subprocess.check_call(['tar', '-xf', app_image_tar_src, '-C', tmp_image_data]))
```

Comment:
i think the `str(` is probably copy paste mistake, right?

Reply:
fixed.